phpunit/class-reusable-blocks-render-test.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Reusuable block rendering tests.
+ *
+ * @package Gutenberg
+ */
+
+/**
+ * Tests reusuable block rendering.
+ */
+class Reusuable_Blocks_Render_Test extends WP_UnitTestCase {
+	/**
+	 * Fake user ID.
+	 *
+	 * @var int
+	 */
+	protected static $user_id;
+
+	/**
+	 * Fake block ID.
+	 *
+	 * @var int
+	 */
+	protected static $block_id;
+
+	/**
+	 * Fake post ID.
+	 *
+	 * @var int
+	 */
+	protected static $post_id;
+
+	/**
+	 * Create fake data before tests run.
+	 *
+	 * @param WP_UnitTest_Factory $factory Helper that creates fake data.
+	 */
+	public static function wpSetUpBeforeClass( $factory ) {
+		self::$user_id = $factory->user->create( array(
+			'role' => 'editor',
+		) );
+
+		self::$post_id = $factory->post->create( array(
+			'post_author'  => self::$user_id,
+			'post_type'    => 'post',
+			'post_status'  => 'publish',
+			'post_title'   => 'Test Post',
+			'post_content' => '<p>Hello world!</p>',
+		) );
+
+		self::$block_id = $factory->post->create( array(
+			'post_author'  => self::$user_id,
+			'post_type'    => 'wp_block',
+			'post_status'  => 'publish',
+			'post_title'   => 'Test Block',
+			'post_content' => '<!-- wp:core/paragraph --><p>Hello world!</p><!-- /wp:core/paragraph -->',
+		) );
+	}
+
+	/**
+	 * Delete fake data after tests run.
+	 */
+	public static function wpTearDownAfterClass() {
+		wp_delete_post( self::$block_id, true );
+		wp_delete_post( self::$post_id, true );
+		self::delete_user( self::$user_id );
+	}
+
+	/**
+	 * Set up.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->registry = new WP_Block_Type_Registry();
+
+		$this->registry->register( 'core/block', array(
+			'attributes'      => array(
+				'ref' => array(
+					'type' => 'number',
+				),
+			),
+			'render_callback' => 'gutenberg_render_block_core_reusable_block',
+		) );
```

Comment:
Could we avoid having to register this both in `blocks/library/block/index.php` and here in the test case?

Perhaps by using `WP_Block_Type_Registry::get_instance()->get_registered()` in the test cases instead of `$this->registry->get_instance()`?

Reply:
Thanks for the review!

Yes, that would have been my preference also, but there are [other existing tests affecting the main registry state](https://github.com/WordPress/gutenberg/blob/84487bab5258fe27897b601914fe5ee339bfcbe8/phpunit/class-dynamic-blocks-render-test.php#L41-L43), and I found that `core/block` was missing whenever this new test ran.

To prevent this from becoming an ongoing issue, I went ahead and straightened this out in the other tests, as an additional commit in this PR. See: https://github.com/WordPress/gutenberg/pull/4608/commits/6e8f9fb8e08f8eb5a70ada582c352f76a8671881